src/streamlink/plugins/mildom.py

```diff
@@ -16,6 +17,30 @@
 log = logging.getLogger(__name__)
 
 
+class MildomHLSStream(HLSStream):
+    __shortname__ = "hls-mildom"
+
+    def __init__(self, session_, url, channel_id=None, quality=None, **args):
+        super().__init__(session_, url, **args)
+        self.session = session_
+        self._url = url
+        self.channel_id = channel_id
+        self.quality = quality
+        self.expiry = time()
+
+    @property
+    def url(self):
+        if self.channel_id and self.quality and time() - self.expiry > 60 * 120:
+            self.expiry = time()
+            for s in Mildom._get_live_streams(self, self.channel_id):
```

Comment:
Creating new stream instances over and over again just to get the updated URL is way too wasteful. The HLSStream should also not make calls back to the Plugin implementation.

Does the `base_url` aka. the `stream_server` even change? Without taking a look, this rather looks like you need to get a new token after a while, which currently gets appended in the `_get_live_streams` method, and not an entire new URL.

The API stuff could all be moved to a separate class that gets shared between the Plugin and HLSStream. Then re-acquiring the token becomes trivial.